academy/lending-protocol/frontend/README.md

```diff
@@ -0,0 +1,292 @@
+# DeFi Lending Platform - Example Application
+
+This is an **example application** designed for **educational purposes** to help developers learn how to interact with the =nil; blockchain using a frontend application. The app demonstrates the process of transacting with =nil;, including transacting with methods and tokens. It also showcases best practices for connecting a frontend application to the blockchain and calling smart contract methods, including handling tokens.
+
+## Purpose
+
+The goal of this application is to give developers an understanding of how to build frontend applications that interact with =nil; and its smart contracts. By going through this example, you can learn the following:
+
+- How to **connect to the =nil; wallet**.
+- How to **read data** from a smart contract.
+- How to **call smart contract methods** for transactions such as lending, borrowing, and repaying, including sending tokens along with method calls via the =nil; wallet.
+- How to structure your frontend code to handle receipts and errors.
+
+### Features
+
+This application allows you to:
+
+- Depositing assets (ETH or USDT) into the protocol
+- Borrowing assets up to 80% of your deposit value
+- Repaying loans with 5% interest (105% of the borrowed amount)
+- Viewing real-time deposit and loan balances
+
+> Note: This example application only supports whole numbers (no decimals) for all transactions.
+
+This repository is designed to provide a simple yet functional template for building decentralized applications (dApps) on top of =nil;.
+
+---
+
+## Pre-requisites
+
+Before running this application, ensure that you have the following:
+
+- **=nil; wallet extension** installed on your browser. You can get it from the [Chrome Web Store](https://chromewebstore.google.com/detail/nil-wallet/kfiailmjchdbjmadbkkldiahpggcjffp?hl=en-GB&utm_source=ext_sidebar).
+- **Node.js** and **npm** installed on your machine.
+- A basic understanding of React, TypeScript, and how smart contracts work.
+
+---
+
+## Installation
+
+### 1. Clone the repository
+
+First, clone this repository to your local machine:
+
+```bash
+git clone https://github.com/NilFoundation/nil
+cd  nil/academy/lending-protocol/frontend
+```
+
+### 2. Install dependencies
+
+Install the necessary dependencies using npm:
+
+```bash
+npm install --legacy-peer-deps
+```
+
+### 3. Configure the RPC URL
+
+Before running the frontend application, you must deploy the lending pool contract and obtain its address.
+
+<strong>Deploy the Lending Pool Contract</strong>
+
+If you haven’t already deployed the contract, navigate to the lending protocol directory and deploy it using Hardhat:
+
+```bash
+cd ../ # Move to the lending protocol directory
+npx hardhat compile
+npx hardhat run-lending-protocol
```

Comment:
this requires: 

- installed `node_modules` for this lending protocol dir package.json
- a globally installed `hardhat`
- defined `.env` in this lending protocol dir